source/dashboard/main.c

```diff
@@ -433,6 +432,94 @@ void heartBeatLED()
     }
 }
 
+static int8_t encoder_count;
+
+void EXTI9_5_IRQHandler(void) {
+    // Check if EXTI9 triggered the interrupt
+    // PD9: Quad Encoder B
+    if (EXTI->PR & EXTI_PR_PR9) { 
+
+        EXTI->PR = EXTI_PR_PR9; // Clear the interrupt pending bit for EXTI9
+
+        // Interrupt handling for quad here
+        encoder_ISR();
+    }
+}
+
+void EXTI15_10_IRQHandler() {
+    // Check if the interrupt was caused by EXTI11
+    if (EXTI->PR & EXTI_PR_PR11) {
+        PHAL_toggleGPIO(PRCHG_LED_GPIO_Port, PRCHG_LED_Pin);
+        SEND_START_BUTTON(q_tx_can, 1);
+
+        // Clear the interrupt pending bit for EXTI11
+        EXTI->PR |= EXTI_PR_PR11;
+    }
+    
+    // Check if EXTI10 triggered the interrupt
+    // PD9: Quad Encoder A
+    if (EXTI->PR & EXTI_PR_PR10) { 
+        // Clear the interrupt pending bit for EXTI10
+        EXTI->PR |= EXTI_PR_PR10; 
+
+        // Interrupt handling for quad here
+        encoder_ISR();
```

Comment:
Maybe call encoder_ISR before you clear the pending bit - if there is another interrupt while you are in this function you might get screwed up, so if the flag is still high so you won't get that issue.